src/Region/Region.cc

```diff
@@ -388,50 +389,94 @@ casacore::LCRegion* Region::GetImageRegion(
-    // points = [p0, p1, p2, p3, center]
-    std::vector<CARTA::Point> points = GetApproximateEllipsePoints(4);
-    points.push_back(center_point);
-    casacore::Vector<casacore::Double> x, y;
+            std::vector<CARTA::Point> points;
+            if (region_type == CARTA::RegionType::ELLIPSE) {
+                // Make polygon with 4 points
+                points = GetApproximateEllipsePoints(4);
+            } else {
+                // Get midpoints of 4 sides of rectangle
+                points = GetRectangleMidpoints();
+            }
 
-    if (ConvertPolygonToImage(points, output_csys, x, y)) {
-        // vector0 is (center, p0), vector1 is (center, p1)
-        auto v0_delta_x = x[0] - x[4];
-        auto v0_delta_y = y[0] - y[4];
-        auto v1_delta_x = x[1] - x[4];
-        auto v1_delta_y = y[1] - y[4];
-
-        auto v0_length = sqrt((v0_delta_x * v0_delta_x) + (v0_delta_y * v0_delta_y));
-        auto v1_length = sqrt((v1_delta_x * v1_delta_x) + (v1_delta_y * v1_delta_y));
-        double converted_length_ratio = v1_length / v0_length;
-
-        // Compare converted to reference length ratio
-        double length_ratio_difference = fabs(ref_length_ratio - converted_length_ratio);
-        spdlog::debug("Ellipse distortion check: length ratio difference={:.3e}", length_ratio_difference);
-        if (length_ratio_difference > 1e-4) {
+            // add center point; points = [p0, p1, p2, p3, center]
+            points.push_back(center_point);
+
+            // convert points to output image pixels
+            casacore::Vector<casacore::Double> x, y;
+            if (ConvertPolygonToImage(points, output_csys, x, y)) {
+                // vector0 is (center, p0), vector1 is (center, p1)
+                auto v0_delta_x = x[0] - x[4];
+                auto v0_delta_y = y[0] - y[4];
+                auto v1_delta_x = x[1] - x[4];
+                auto v1_delta_y = y[1] - y[4];
+
+                auto v0_length = sqrt((v0_delta_x * v0_delta_x) + (v0_delta_y * v0_delta_y));
+                auto v1_length = sqrt((v1_delta_x * v1_delta_x) + (v1_delta_y * v1_delta_y));
+                double converted_length_ratio = v1_length / v0_length;
+
+                // Compare converted to reference length ratio
+                double length_ratio_difference = fabs(ref_length_ratio - converted_length_ratio);
+                spdlog::debug("{} distortion check: length ratio difference={:.3e}", RegionName(region_type), length_ratio_difference);
+                if (length_ratio_difference < 1e-4) {
+                    // Passed ratio check; check dot product
+                    double converted_dot_product = (v0_delta_x * v1_delta_x) + (v0_delta_y * v1_delta_y);
+                    spdlog::debug("{} distortion check: dot product={:.3e}", RegionName(region_type), converted_dot_product);
+
+                    if (fabs(converted_dot_product) < 1e-4) {
```

Comment:
@pford Thanks for preparing the handy debug log for testing. I have done some tests with rectangle and ellipse both unrotated and rotated and focused on the threshold of the dot product check. By experimenting the different threshold, **1e-2** (no idea why we need a larger threshold here🙈 Maybe this is due to imageA_pixel->imageA-world->imageB_world->imageB_pixel conversion errors?) works properly without any inconsistent statistics as far as I can try and see.

Reply:
I suspect it is due to world->pixel conversion using wcslib wcss2p().  Working on using ast for this conversion for issue #541 .